quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -172,13 +190,48 @@ impl TypedSourceFactory for FileSourceFactory {
                 current_offset: offset as u64,
                 num_lines_processed: 0,
             },
-            reader: BufReader::new(reader),
+            reader,
             params,
         };
         Ok(file_source)
     }
 }
 
+struct FileSourceReader {
+    reader: BufReader<Box<dyn AsyncRead + Send + Unpin>>,
+    num_bytes_to_skip: usize,
+}
+
+impl FileSourceReader {
+    fn new(reader: Box<dyn AsyncRead + Send + Unpin>, num_bytes_to_skip: usize) -> Self {
+        Self {
+            reader: BufReader::new(reader),
+            num_bytes_to_skip,
+        }
+    }
+
+    async fn skip(&mut self) -> io::Result<()> {
```

Comment:
```suggestion
    // This function is only called for GZIP file.
    // Because they cannot be seeked into, we have to scan them to the right initial position.
    async fn skip(&mut self) -> io::Result<()> {
```